tests/functional/func_lpf_put_parallel_bad_pattern.cpp

```diff
@@ -28,74 +28,73 @@ void spmd( lpf_t lpf, lpf_pid_t pid, lpf_pid_t nprocs, lpf_args_t args)
     const unsigned n = sqrt(nprocs);
     unsigned i;
     unsigned * xs, *ys;
-    ys = malloc( sizeof(ys[0]) * n);
-    xs = malloc( sizeof(xs[0]) * n);
+    ys = (unsigned *) malloc( sizeof(ys[0]) * n);
+    xs = (unsigned *) malloc( sizeof(xs[0]) * n);
     for (i = 0; i < n; ++i)
     {
         xs[i] = i;
         ys[i] = 0;
     }
         
     rc = lpf_resize_message_queue( lpf, n);
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
     rc = lpf_resize_memory_register( lpf, 2 );
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
     rc = lpf_sync( lpf, LPF_SYNC_DEFAULT );
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
  
     lpf_memslot_t xslot = LPF_INVALID_MEMSLOT;
     lpf_memslot_t yslot = LPF_INVALID_MEMSLOT;
     rc = lpf_register_local( lpf, xs, sizeof(xs[0]) * n, &xslot );
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
     rc = lpf_register_global( lpf, ys, sizeof(ys[0]) * n, &yslot );
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
 
     rc = lpf_sync( lpf, LPF_SYNC_DEFAULT);
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
 
     // Check that data is OK.
     for (i = 0; i < n; ++i)
     {
-        EXPECT_EQ( "%u", i, xs[i] );
-        EXPECT_EQ( "%u", 0u, ys[i] );
+        EXPECT_EQ( i, xs[i] );
+        EXPECT_EQ( 0u, ys[i] );
     }
 
     if ( pid < n )
     {
         for ( i = 0; i < n; ++ i)
         {
-            EXPECT_LT("%u", i*n, nprocs);
+            EXPECT_LT( i*n, nprocs);
             rc = lpf_put( lpf, xslot, sizeof(xs[0])*i, 
                     i*n, yslot, sizeof(ys[0])*pid, sizeof(xs[0]), 
                     LPF_MSG_DEFAULT );
-            EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+            EXPECT_EQ( LPF_SUCCESS, rc );
         }
     }
 
         
     rc = lpf_sync( lpf, LPF_SYNC_DEFAULT );
-    EXPECT_EQ( "%d", LPF_SUCCESS, rc );
+    EXPECT_EQ( LPF_SUCCESS, rc );
 
     for (i = 0; i < n; ++i)
     {
-        EXPECT_EQ( "%u", i, xs[i] );
+        EXPECT_EQ( i, xs[i] );
         if ( pid % n == 0 && pid < n*n)
-            EXPECT_EQ( "%u", pid / n, ys[i] );
+            EXPECT_EQ( pid / n, ys[i] );
         else
-            EXPECT_EQ( "%d", 0, ys[i] );
+            EXPECT_EQ( 0, ys[i] );
     }
 
 }
 
 /** 
  * \test Test lpf_put by doing a pattern which bad for a sparse all-to-all
- * \pre P >= 1024
- * \pre P <= 1024
+ * \pre P >= 5
+ * \pre P <= 5
  * \return Exit code: 0
```

Comment:
Why this rather severe reduction in P?

Reply:
The intended behaviour was that it runs using the max number of LPF ranks that could reasonably be used to run the test with. This behaviour is now restored.